picard/coverart/providers/caa.py

```diff
@@ -94,34 +98,203 @@ def caa_url_fallback_list(desired_size, thumbnails):
+        if types_include is None:
+            types_include = []
+        if types_exclude is None:
+            types_exclude = []
 
         self.setWindowTitle(_("Cover art types"))
-        self._items = {}
         self.layout = QtWidgets.QVBoxLayout(self)
 
+        # Create list boxes for dialog
+        self.list_include = ListBox()
+        self.list_exclude = ListBox()
+        self.list_ignore = ListBox()
+
+        # Populate list boxes from current settings
+        self.fill_lists(types_include, types_exclude)
+
+        # Set triggers when the lists receive the current focus
+        self.list_include.clicked.connect(partial(self.clear_focus, [self.list_ignore, self.list_exclude,]))
+        self.list_exclude.clicked.connect(partial(self.clear_focus, [self.list_ignore, self.list_include,]))
+        self.list_ignore.clicked.connect(partial(self.clear_focus, [self.list_include, self.list_exclude,]))
+
+        # Add instructions to the dialog box
+        instructions = QtWidgets.QLabel()
+        instructions.setText(_("Please select the contents of the image type 'Include' and 'Exclude' lists."))
+        instructions.setWordWrap(True)
+        instructions.setSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Expanding)
+        self.layout.addWidget(instructions)
+
         grid = QtWidgets.QWidget()
         gridlayout = QtWidgets.QGridLayout()
         grid.setLayout(gridlayout)
 
-        for index, caa_type in enumerate(CAA_TYPES):
-            row = index // self._columns
-            column = index % self._columns
-            name = caa_type["name"]
-            text = translate_caa_type(name)
-            item = QtWidgets.QCheckBox(text)
-            item.setChecked(name in types)
-            self._items[item] = caa_type
-            gridlayout.addWidget(item, row, column)
+        row = 0
+        column = 0
+        item = QtWidgets.QLabel()
+        item.setText(_("Include types list"))
+        gridlayout.addWidget(item, row, column)
+
+        column = 4
+        item = QtWidgets.QLabel()
+        item.setText(_("Exclude types list"))
+        gridlayout.addWidget(item, row, column)
+
+        row = 1
+        column = 0
+        gridlayout.addWidget(self.list_include, row, column)
+
+        column = 1
+        self.arrows_include = ArrowsColumn(
+            self.list_include,
+            self.list_ignore,
+            callback=self.set_buttons_enabled_state,
+        )
+        gridlayout.addWidget(self.arrows_include, row, column)
+
+        column = 2
+        gridlayout.addWidget(self.list_ignore, row, column)
+
+        column = 3
+        self.arrows_exclude = ArrowsColumn(
+            self.list_exclude,
+            self.list_ignore,
+            callback=self.set_buttons_enabled_state,
+            reverse=True
+        )
+        gridlayout.addWidget(self.arrows_exclude, row, column)
+
+        column = 4
+        gridlayout.addWidget(self.list_exclude, row, column)
```

Comment:
Code would be perhaps easier to read using a small local function, i was thinking about something like:

```python
        self.arrows_include = ArrowsColumn(
            self.list_include,
            self.list_ignore,
            callback=self.set_buttons_enabled_state,
        )

        self.arrows_exclude = ArrowsColumn(
            self.list_exclude,
            self.list_ignore,
            callback=self.set_buttons_enabled_state,
            reverse=True
        )

        def add_widget(row=0, column=0, widget):
             gridlayout.addWidget(widget, row, column)

        add_widget(row=0, column=0, QtWidgets.QLabel(_("Include types list")))
        add_widget(row=1, column=0, self.list_include)

        add_widget(row=1, column=1, self.arrows_include)
        add_widget(row=1, column=2, self.list_ignore)
        add_widget(row=1, column=3, self.arrows_include)

        add_widget(row=0, column=4, QtWidgets.QLabel(_("Exclude types list")))
        add_widget(row=1, column=4, self.list_exclude)
```

No strong opinion here though.

Reply:
I like it.  It definitely looks cleaner than the way it is currently.